apps/aefate/src/aefa_fate_op.erl

```diff
@@ -1247,6 +1261,134 @@ spend(Arg0, Arg1, ES0) ->
+
+clone_g(Arg0, Arg1, Arg2, Arg3, Arg4, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {[?FATE_CONTRACT(CloneePK), InitArgsTypes, Value, GasCap, Prot], ES1} =
+        get_op_args([Arg0, Arg1, Arg2, Arg3, Arg4], ES0),
+    clone_common(CloneePK, InitArgsTypes, Value, {gas_cap, GasCap}, Prot, ES1).
+
+clone_common(CloneePK, InitArgsTypes, Value, GasCap, Prot, ES0) ->
+    Protected =
+        case Prot of
+            false -> unprotected;
+            true  -> protected;
+            _     -> aefa_fate:abort({value_does_not_match_type, Prot, bool}, ES0)
+        end,
+
+    {ok, FinalCloneePK} =
+        aefa_engine_state:contract_find_final_ref(CloneePK, ES0),
+
+    {ContractPK, ES1} = create_contract({clone, FinalCloneePK}, Value, ES0),
+    ReplaceInitResult
+        = fun(ES0_) ->
+                  %% init returns unit, so we get rid of it
+                  %% if the call was protected, rewrap the result
+                  case Protected of
+                      unprotected ->
+                          {{tuple, {}}, ES1_} = get_op_arg({stack, 0}, ES0_),
+                          write({stack, 0}, ?FATE_CONTRACT(ContractPK), ES1_);
+                      protected ->
+                          case get_op_arg({stack, 0}, ES0_) of
+                              {{variant,[0,1],1,{{tuple,{}}}}, ES1_} ->
+                                  write( {stack, 0}, make_some(?FATE_CONTRACT(ContractPK))
+                                       , ES1_
+                                       );
+                              {{variant, [0,1],0,{}}, ES1_} ->
+                                  write({stack, 0}, make_none(), ES1_)
+                          end
+                  end
+          end,
+    case remote_call_common(
+           ?FATE_CONTRACT(ContractPK),
+           init,
+           InitArgsTypes,
+           ?FATE_TYPEREP({tuple, []}),
+           0,
+           GasCap,
+           Protected,
+           ReplaceInitResult,
+           ES1)
+    of
+        {ok, ES2} ->
+            {jump, 0, ES2};
+        {failed_protected_call, ES2} ->
+            ES3 = aefa_engine_state:remove_contract(ContractPK, ES2),
+            {next, write({stack, 0}, make_none(), ES3)}
+    end.
+
+create_contract(CodeOrPK, Amount, ES0) ->
+    Current = aefa_engine_state:current_contract(ES0),
+
+    AS0              = aefa_engine_state:chain_api(ES0),
+    {ok, Nonce, AS1} = aefa_chain_api:next_nonce(Current, AS0),
+
+    Contract0  =
+        case CodeOrPK of
+            {new, Code} -> aect_contracts:new(
+                              Current, Nonce,
+                              #{vm => aefa_engine_state:vm_version(ES0)
+                              , abi => ?ABI_FATE_SOPHIA_1
+                              }, Code, 0);
+            {clone, PK} -> aect_contracts:new_clone(
+                             Current, Nonce,
+                             #{vm => aefa_engine_state:vm_version(ES0)
```

Comment:
Doesn't this potentially change the VM version of a contract? For now we only allow to clone FATEv2 contracts, and any contract doing clone will also be on FATEv2. But in the future if there is a FATEv3, etc?

Reply:
In the future I would assert that the version is the same as caller's

Reply:
I would assume the VM-version should be the same as the cloned contract originally have?!

Reply:
done